addons/mail/static/src/core/common/thread_model.js

```diff
@@ -845,6 +845,7 @@ export class Thread extends Record {
         ]);
         if (this.selfMember) {
             this.selfMember.seen_message_id = this.newestPersistentNotEmptyOfAllMessage;
+            this.selfMember.new_message_separator = this.newestPersistentNotEmptyOfAllMessage;
```

Comment:
If we code the feature better, the non empty notion should become irrelevant.
Compute of the separator is currently too linked to the display of the messages.